pkg/prometheus/promcfg.go

```diff
@@ -731,6 +732,39 @@ func (cg *ConfigGenerator) addProxyConfigtoYaml(
 	return cfg
 }
 
+func (cg *ConfigGenerator) addCustomHTTPConfigtoYaml(
+	cfg yaml.MapSlice,
+	store assets.StoreGetter,
+	customHTTPConfig monitoringv1.CustomHTTPConfig,
+) yaml.MapSlice {
+	if reflect.ValueOf(customHTTPConfig).IsZero() {
+		return cfg
+	}
+
+	if len(customHTTPConfig.HTTPHeaders) == 0 {
+		return cfg
+	}
+
+	cgCustomHTTPConfig := cg.WithMinimumVersion("2.55.0")
+	httpHeaders := yaml.MapSlice{}
+
+	for _, header := range customHTTPConfig.HTTPHeaders {
+		httpHeader := yaml.MapSlice{}
+		var secrets []string
+		for _, ref := range header.SecretRefs {
+			value, _ := store.GetSecretKey(ref)
+			secrets = append(secrets, string(value))
+		}
+		if len(secrets) > 0 {
+			httpHeader = append(httpHeader, yaml.MapItem{Key: "secrets", Value: secrets})
+		}
+
+		httpHeaders = append(httpHeaders, yaml.MapItem{Key: http.CanonicalHeaderKey(header.Name), Value: httpHeader})
```

Comment:
what if we have 2 header keys with different cases (e.g. "X-Foo" and "x-foo"). Will it break?

Reply:
No. All header keys (e.g. "X-Foo" and "x-foo") added to `httpHeaders` config. 
@simonpasquier  WDYT?